utils/utils.go

```diff
@@ -0,0 +1,42 @@
+package utils
```

Comment:
these functions should be either moved under `internal` folder or made unexportable. From library perspective they probably are not needed to used